ProjFS.Mac/PrjFSKext/VirtualizationRoots.cpp

```diff
@@ -287,10 +287,10 @@ static VirtualizationRootHandle FindUnusedIndexOrGrow_Locked()
     if (RootHandle_None == rootIndex)
     {
         // No space, resize array
-        uint16_t newLength = MIN(s_maxVirtualizationRoots * 2u, INT16_MAX + 1u);
-        if (newLength <= s_maxVirtualizationRoots)
+        uint16_t newLength;
+        if (__builtin_mul_overflow(s_maxVirtualizationRoots, 2, &newLength))
         {
-            assert(newLength > 0);
```

Comment:
This check had nothing to do with overflow, but catching the situation where `s_maxVirtualizationRoots` was not initialised. Happened to me in unit tests, which caused things to screw up. Please don't remove the assert.

Reply:
Bonus points: as it's evidently not obvious why the `assert` is there, perhaps turn it into an `assertf` with a suitable message.

Reply:
I convinced myself the math would never overflow, removed the overflow checking, and made the assert more descriptive.